rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/normalize.scala

```diff
@@ -722,17 +734,52 @@ object ProcNormalizeMatcher {
         } yield ProcVisitOutputs(
           input.par.prepend(
             Send(
-              nameMatchResult.chan,
+              nameMatchResult.par,
               dataResults._1,
               persistent,
               ParLocallyFree
-                .locallyFree(nameMatchResult.chan, input.env.depth) | dataResults._3,
-              ParLocallyFree.connectiveUsed(nameMatchResult.chan) || dataResults._4
+                .locallyFree(nameMatchResult.par, input.boundMapChain.depth) | dataResults._3,
+              ParLocallyFree.connectiveUsed(nameMatchResult.par) || dataResults._4
             )
           ),
-          dataResults._2.knownFree
+          dataResults._2.freeMap
         )
 
+      case p: PSendSynch =>
+        val identifier = UUID.randomUUID().toString
+        val nameVar    = new NameVar(identifier)
+
+        val send: PSend = {
+          p.listproc_.prepend(new PEval(nameVar))
```

Comment:
I think I found that the new var can only be the first one because of this

Reply:
I think most of the existing libraries put this `Ack` name in the tail not the head.
But I am not sure which one is better. Just curious whether it is a problem to put it in the tail.

Reply:
If it goes in the tail, we can no longer use remainder syntax. Do you see why?